scs2-simulation/src/main/java/us/ihmc/scs2/simulation/robot/multiBodySystem/SimFloatingRootJoint.java

```diff
@@ -44,97 +50,167 @@ public SimFloatingRootJoint(String name, SimRigidBodyBasics predecessor, RigidBo
+                                                          TwistReadOnly twist,
+                                                          Vector3DReadOnly linearAccelerationOriginal,
+                                                          Vector3DBasics linearAccelerationTransformed)
+   {
+      quaternion.inverseTransform(linearAccelerationOriginal, linearAccelerationTransformed);
+      MecanoTools.addCrossToVector(twist.getLinearPart(), twist.getAngularPart(), linearAccelerationTransformed);
+   }
+
+   private static void transformLinearAcceleration(QuaternionReadOnly quaternion,
+                                                   TwistReadOnly twist,
+                                                   Vector3DReadOnly linearAccelerationOriginal,
+                                                   Vector3DBasics linearAccelerationTransformed)
+   {
+      linearAccelerationTransformed.set(linearAccelerationOriginal);
+      MecanoTools.addCrossToVector(twist.getAngularPart(), twist.getLinearPart(), linearAccelerationTransformed);
+      quaternion.transform(linearAccelerationTransformed);
+   }
+
+   private static void buildOrientationListeners(YoFrameQuaternion quaternion, YoFrameYawPitchRoll yawPitchRoll)
+   {
       MutableBoolean updatingYPR = new MutableBoolean(false);
       MutableBoolean updatingQuat = new MutableBoolean(false);
 
-      jointQuaternion.getYoQs().addListener(v ->
+      quaternion.getYoQs().addListener(v ->
+                                       {
+                                          if (updatingQuat.booleanValue())
+                                             return;
+
+                                          updatingYPR.setTrue();
+                                          try
+                                          {
+                                             yawPitchRoll.set(quaternion);
+                                          }
+                                          finally
+                                          {
+                                             updatingYPR.setFalse();
+                                          }
+                                       });
+
+      double updateThreshold = 1.0e-6;
+
+      // Doing 1 component at a time to avoid updating the quaternion with old data.
+      yawPitchRoll.getYoYaw().addListener(v ->
+                                          {
+                                             if (updatingYPR.booleanValue())
+                                                return;
+                                             updatingQuat.setTrue();
+                                             try
+                                             {
+                                                if (!EuclidCoreTools.epsilonEquals(quaternion.getYaw(), yawPitchRoll.getYaw(), updateThreshold))
+                                                   quaternion.setYawPitchRoll(yawPitchRoll.getYaw(), quaternion.getPitch(), quaternion.getRoll());
```

Comment:
The idea is to only update using the 1 variable that is changing, the other 2 variables maybe outdated